server/src/internalClusterTest/java/org/opensearch/snapshots/DeleteSnapshotIT.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.snapshots;
+
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.util.FeatureFlags;
+import org.opensearch.test.FeatureFlagSetter;
+import org.opensearch.test.OpenSearchIntegTestCase;
+
+import java.nio.file.Path;
+
+import static org.opensearch.test.hamcrest.OpenSearchAssertions.assertAcked;
+
+@OpenSearchIntegTestCase.ClusterScope(scope = OpenSearchIntegTestCase.Scope.TEST, numDataNodes = 0)
+public class DeleteSnapshotIT extends AbstractSnapshotIntegTestCase {
+
+    public void testDeleteSnapshot() throws Exception {
+        disableRepoConsistencyCheck("Remote store repository is being used in the test");
+        FeatureFlagSetter.set(FeatureFlags.REMOTE_STORE);
+        internalCluster().startClusterManagerOnlyNode();
+        internalCluster().startDataOnlyNode();
+
+        final String snapshotRepoName = "snapshot-repo-name";
+        final Path snapshotRepoPath = randomRepoPath();
+        createRepository(snapshotRepoName, "fs", snapshotRepoPath);
+
+        final Path remoteStoreRepoPath = randomRepoPath();
+        final String remoteStoreRepoName = "remote-store-repo-name";
+        createRepository(remoteStoreRepoName, "fs", remoteStoreRepoPath);
+
+        final String indexName = "index-1";
+        createIndexWithRandomDocs(indexName, randomIntBetween(5, 10));
+
+        final String remoteStoreEnabledIndexName = "remote-index-1";
+        final Settings remoteStoreEnabledIndexSettings = getRemoteStoreBackedIndexSettings(remoteStoreRepoName);
+        createIndex(remoteStoreEnabledIndexName, remoteStoreEnabledIndexSettings);
+        indexRandomDocs(remoteStoreEnabledIndexName, randomIntBetween(5, 10));
+
+        final String snapshot = "snapshot";
+        createFullSnapshot(snapshotRepoName, snapshot);
+        assert (getLockFilesInRemoteStore(remoteStoreEnabledIndexName, remoteStoreRepoName).length == 0);
+        assert (getRepositoryData(snapshotRepoName).getSnapshotIds().size() == 1);
+
+        assertAcked(startDeleteSnapshot(snapshotRepoName, snapshot).get());
+        assert (getRepositoryData(snapshotRepoName).getSnapshotIds().size() == 0);
+    }
+
+    public void testDeleteShallowCopySnapshot() throws Exception {
```

Comment:
can we add some more tests:
1. With deleting multiple shallow snapshots as part of single delete call and repo having only shallow snapshot
2.With deleting multiple shallow snapshots as part of single delete call but with both partial and full snapshot present in the repo 
3.With deleting multiple full snapshots as part of single delete call but with both partial and full snapshot present in the repo 
4. Deleting both full and shallow snapshot in single request with only subset of snapshots to be deleted not all.
5. Deleting all the full and shallow snapshot in single request present in the repository.

Reply:
@sohami  , @kasundra07  : Is there a way we can restrict toggling `remote_store_index_shallow_copy`  setting ? I was thinking if we could restrict this to only new repositories and make it read only , it would make our mental model quite simple. We would not need to think of this getting toggled every now and then at every place.  

On a side node, we are thinking of having a read only repository for remote store. That repository can't be deleted/modified and would be created at bootstrap time via yml . We can use the constructs there if we want to achieve it .

Reply:
lets add test for lock file deletion and `dat` file deletion failures as well to ensure deletion really fails.

Reply:
As discussed, will add the deletion failure IT in the followup PR. Have manually tested it for now.